clients/da-vinci-client/src/test/java/com/linkedin/davinci/store/rocksdb/RocksDBSstFileWriterTest.java

```diff
@@ -255,6 +259,124 @@ public void testOpenWithAllValidFiles() throws IOException {
+  public void testSyncWithInCorrectChecksum() throws IOException, RocksDBException {
+    RocksDBSstFileWriter rocksDBSstFileWriter = null;
+    try {
+      rocksDBSstFileWriter = new RocksDBSstFileWriter(
+          STORE_NAME,
+          PARTITION_ID,
+          "",
+          new EnvOptions(),
+          new Options(),
+          DB_DIR,
+          IS_RMD,
+          ROCKS_DB_SERVER_CONFIG);
+      Map<String, String> checkpointedInfo = new HashMap<>();
+
+      // Checkpoint that 1 sst file should be found
+      checkpointedInfo.put(rocksDBSstFileWriter.getLastCheckPointedSSTFileNum(), "0");
+      // create 10 sst file
+      createSstFiles(5);
+
+      rocksDBSstFileWriter.open(checkpointedInfo, Optional.of(() -> {
+        CheckSum sstFileFinalCheckSum = CheckSum.getInstance(CheckSumType.MD5);
+        sstFileFinalCheckSum.update("wrong_key".getBytes());
+        sstFileFinalCheckSum.update("wrong_value".getBytes());
+        return sstFileFinalCheckSum.getCheckSum();
+      }));
+      // check that only 2 files exists ("0" after cleanup by checkDatabaseIntegrity)
+      // and "1" after rocksDBSstFileWriter.open opens a new file
+      Assert.assertEquals(getNumberOfFilesInTempDirectory(), 2);
+      rocksDBSstFileWriter.put("key".getBytes(), ByteBuffer.wrap("value".getBytes()));
+      // call sync to verify checksum
+      rocksDBSstFileWriter.sync();
+    } finally {
+      if (rocksDBSstFileWriter != null) {
+        rocksDBSstFileWriter.close();
+      }
+      deleteTempDatabaseDir();
+    }
+  }
+
+  @Test(expectedExceptions = VeniceException.class, expectedExceptionsMessageRegExp = "Checksum mismatch in SST files.*")
+  public void testSyncWithMissingFile() throws IOException, RocksDBException {
+    RocksDBSstFileWriter rocksDBSstFileWriter = null;
+    try {
+      rocksDBSstFileWriter = new RocksDBSstFileWriter(
+          STORE_NAME,
+          PARTITION_ID,
+          "",
+          new EnvOptions(),
+          new Options(),
+          DB_DIR,
+          IS_RMD,
+          ROCKS_DB_SERVER_CONFIG);
+      Map<String, String> checkpointedInfo = new HashMap<>();
+
+      // Checkpoint that 6 sst file should be found
+      checkpointedInfo.put(rocksDBSstFileWriter.getLastCheckPointedSSTFileNum(), "5");
+      // create 10 sst file
+      createSstFiles(10);
+
+      rocksDBSstFileWriter.open(checkpointedInfo, Optional.of(() -> "1".getBytes()));
+      // check that only 7 files exists (0-5 after cleanup by checkDatabaseIntegrity)
+      // and 6 after rocksDBSstFileWriter.open opens a new file
+      Assert.assertEquals(getNumberOfFilesInTempDirectory(), 7);
+      rocksDBSstFileWriter.put("key".getBytes(), ByteBuffer.wrap("value".getBytes()));
+      deleteAllSstFiles(getNumberOfFilesInTempDirectory());
+      // call sync after deleting all the files to mimic exception thrown during graceful shutdown of servers with
```

Comment:
Other change looks good, but I am confused by the intention of this particular test. Could you explain a bit more about this one?

Reply:
This was the RC behind this test failing after enabling checksum calculation. so added a test to verify that the checksum fails if there are no sst files during the sync() which is expected.